modules/core/04-channel/v2/keeper/msg_server.go

```diff
@@ -63,12 +63,24 @@ func (k *Keeper) RegisterCounterparty(goCtx context.Context, msg *channeltypesv2
 // SendPacket defines a rpc handler method for MsgSendPacket.
 func (k *Keeper) SendPacket(ctx context.Context, msg *channeltypesv2.MsgSendPacket) (*channeltypesv2.MsgSendPacketResponse, error) {
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
 	sequence, destChannel, err := k.sendPacket(ctx, msg.SourceChannel, msg.TimeoutTimestamp, msg.Payloads)
 	if err != nil {
 		sdkCtx.Logger().Error("send packet failed", "source-channel", msg.SourceChannel, "error", errorsmod.Wrap(err, "send packet failed"))
 		return nil, errorsmod.Wrapf(err, "send packet failed for source id: %s", msg.SourceChannel)
 	}
 
+	// Note, the validate basic function in sendPacket does the timeoutTimestamp != 0 check and other stateless checks on the packet.
+	// timeoutTimestamp must be less than current block time + MaxTimeoutDelta
+	if msg.TimeoutTimestamp > uint64(sdkCtx.BlockTime().Unix())+channeltypesv2.MaxTimeoutDelta {
+		return nil, errorsmod.Wrap(channeltypesv2.ErrMaxTimeoutDeltaExceeded, "timeout exceeds the maximum expected value")
+	}
+
+	// timeoutTimestamp must be greater than current block time
+	if uint64(sdkCtx.BlockTime().Unix()) > msg.TimeoutTimestamp {
+		return nil, errorsmod.Wrap(channeltypesv2.ErrTimeoutTooLow, "timeout is less than the current block timestamp")
+	}
```

Comment:
since sdk block time is a `time.Time`, we could leverage the `time.After()` / `time.Before()` apis from std lib here, by creating a `time.Time` with the absolute timeout. Would imagine it to be something like this:

```golang
// ------------------------------------------------
const timeoutDelta time.Duration = 24 * time.Hour
// ------------------------------------------------

timeout := time.Unix(int64(msg.TimeoutTimestamp), 0)
if timeout.Before(sdkCtx.BlockTime()) {
	// err timeout less than current block timestamp
}

if timeout.After(sdkCtx.BlockTime().Add(timeoutDelta)) {
	// err timeout exceeds max exp value
}
```

I'm fine with using uint64 absolute values too. But it might look a little less verbose to leverage the stdlib apis. Interested on input from others!

Reply:
i think we should use the time package as well when possible

Reply:
Find out that using the time package is the idiomatic choice in go. My bad. Changes applied as suggested!